raving-remoras/bangazon-site
# Rd remove product

# Description
Add description about what this pull request covers.


Given the user is authenticated
When the user clicks on the My Products hyperlink in the navigation bar
Then the user will be presented with a list of products they (s)he has posted for sale

Given the list of products is presented to the user
When the user clicks the Delete button next to a posted product
Then the product for sale is removed from the system

Given a user is authenticated
When a user clicks the My Products hyperlink in the navigation bar
Then the user will be presented with product status for each product uploaded including "current inventory" and "number sold"

Given a user is not authenticated
Then the My Products hyperlink will not be available in the navigation bar

## Related Ticket(s)
#16, #20

## Steps to Test Solution

1. Make sure you can't access the My Products view (/my_products) when not authenticated
2. Check to make sure that the counts of sold and available items on My Products page reflect your DB
3. No delete button should be available if an item is sold out
4. Clicking delete should bring you to a new view-- from there, cancel should bring you back to the my products list. Delete should bring you back to the list as well, and the item you removed should not be there. 
5. Delete something that is in someone else's cart without a payment type-- those join entities should be deleted.

## Testing


## Documentation
